src/linear_algebra/matrix/nonsingular_inverse.lean

```diff
@@ -383,23 +383,101 @@ begin
     apply invertible_of_det_invertible, },
 end
 
-lemma is_unit_det_of_left_inverse (B : matrix n n α) (h : B ⬝ A = 1) : is_unit A.det :=
+/- `is_unit_of_invertible A`
+   converts the "stronger" condition `invertible A` to proposition `is_unit A`. -/
+
+/- `matrix.is_unit_det_of_invertible` converts `invertible A` to `is_unit A.det`. -/
+lemma is_unit_det_of_invertible [invertible A] : is_unit A.det :=
+@is_unit_of_invertible _ _ _(det_invertible_of_invertible A)
+
+@[simp]
+lemma inv_eq_nonsing_inv_of_invertible [invertible A] : ⅟ A = A⁻¹ :=
+begin
+  suffices : is_unit A,
+  { rw [←this.mul_left_inj, inv_of_mul_self, matrix.mul_eq_mul, nonsing_inv_mul],
+    rwa ←is_unit_iff_is_unit_det },
+  exact is_unit_of_invertible _
+end
+
+variables {A} {B}
+
+/-- "Lift" the proposition `is_unit A.det` to a constructive inverse of `A`. -/
+noncomputable def invertible_of_is_unit_det  (h : is_unit A.det) : invertible A :=
+⟨A⁻¹, nonsing_inv_mul A h, mul_nonsing_inv A h⟩
+
+/-- "Lift" the proposition `is_unit A` to a constructive inverse of `A`. -/
+noncomputable def invertible_of_is_unit  (h' : is_unit A) : invertible A :=
+by have h := (is_unit_iff_is_unit_det A).1 h';
+exact ⟨A⁻¹, nonsing_inv_mul A h, mul_nonsing_inv A h⟩
```

Comment:
Is this different to `is_unit.invertible`?

Reply:
#check is_unit.invertible reports unknown identifier. I am not sure what you refer to.

Reply:
Then you need to run `git fetch origin; git merge origin/master`, as [`is_unit.invertible` definitely exists](https://leanprover-community.github.io/mathlib_docs/algebra/invertible.html#is_unit.invertible)

Reply:
Good point! They are the same. I change the def to a doc for the user.